extensions/mssql/src/sqlToolsServer.ts

```diff
@@ -117,12 +117,16 @@ export class SqlToolsServer {
 	}
 }
 
-async function generateServerOptions(logPath: string, executablePath: string): Promise<ServerOptions> {
+function generateServerOptions(logPath: string, executablePath: string): ServerOptions {
 	const launchArgs = getCommonLaunchArgsAndCleanupOldLogFiles(logPath, 'sqltools.log', executablePath);
-	const enableAsyncMessageProcessing = await getParallelMessageProcessingConfig();
+	const enableAsyncMessageProcessing = getParallelMessageProcessingConfig();
 	if (enableAsyncMessageProcessing) {
 		launchArgs.push('--parallel-message-processing');
 	}
+	const enableSqlAuthenticationProvider = getEnableSqlAuthenticationProviderConfig();
```

Comment:
Actually - is this something that can be toggled during runtime? Because STS already gets notified when settings are changed (https://github.com/microsoft/sqltoolsservice/blob/main/src/Microsoft.SqlTools.ServiceLayer/Workspace/WorkspaceService.cs#L326) so you could just hook into that instead and then remove the need for the app restart. That doesn't get the app quality - but that'd be easy enough to pass up as a parameter instead of this.

Reply:
I don't think it's a safe toggle, since it'll also change the way Access tokens are sent to STS and refreshed, it will make existing connections unstable. I would rather force a reload when this is enabled.